server/game/cards/attachments/03/needle.js

```diff
@@ -8,13 +8,18 @@ class Needle extends DrawCard {
 
         this.interrupt({
             when: {
-                onSacrificed: event => event.card === this.parent
+                onSacrificed: event => {
+                    if(event.card === this.parent) {
+                        this.parentCard = this.parent;
+                        return true;
+                    }
```

Comment:
Should explicitly return false when it doesn't go into the `if`. Also don't especially like this pattern of storing off the parent card in the event handler. This should be framework automated, either by the cost system or (more likely) the event system.

In this case, it's unnecessary anyway - you can get it off of `context.event.card` in the handler.